### DISCUSSION TITLE
Target Cadence

### DISCUSSION THREAD

**Discussion by steffidortmund** [dvmarinoff/Auuki#59] [Ideas]
Hi, in .zwo workouts you can also define a target cadence. What do you think about adding this here?

**Comment by dvmarinoff** [comment]
Yep, no problem, I've been thinking about this feature, but it's just that I don't train with cadence targets. I will implement this after I get the development branch more stable (maybe start of December).

Meanwhile can you describe how do you use cadence targets, so that I can get an idea?

**Comment by steffidortmund** [comment]
That's cool! I attached an example of a workout with defined cadence, I think it should become clear there:
<img width="564" alt="workout" src="https://user-images.githubusercontent.com/66872604/144088515-df804958-e82d-4101-b00e-cbf5c5d8c026.PNG">

So I think it is just another value of the zwo-file next to the current cadence to compare, if a cadence is defined in the workout (at the moment I am using https://zwofactory.com/ to create workouts) 

One other thing.. when I used Flux I have missed, that the heart rate wasn't saved in the fit file. Is it planned to add this? :)

And by the way.. I really like your project!! :) It is exactly what I was looking for!

**Reply by dvmarinoff** [reply]
Hey, thanks for catching that bug with the Heart Rate data! I am working on it hopefully will get it resolved very soon. Here is the [Issue](https://github.com/dvmarinoff/Flux/issues/61) that will track the progress. I rewrote the whole .FIT section recently and it seems that somehow I've managed to brake the Heart Rate data.

**Reply by steffidortmund** [reply]
Thank you!!

**Comment by dvmarinoff** [comment]
As for the Cadence targets the zwo parser was maybe one of the first pieces of code I added to the project and it was just a placeholder implementation that was there to do just the very basics. Last week I rewrote the whole thing so now it can not just read, but also write zwo files, which will allow for a future editing directly in the app, and I added a test suite. It is also collecting the cadence targets in the file. Now with the ground work done, I need to figure out some kind of user interface for the cadence target, and also where to place it inside the app.

As far as I understand the cadence target is just an additional one, that goes along the main power target, and it doesn’t change without the power target also changing, right?

**Reply by steffidortmund** [reply]
It depends on what you want and how you create your workout :) You can also have the same power target and different cadence targets:
![image](https://user-images.githubusercontent.com/66872604/144288097-20a1bcd7-122a-4787-b2e0-446e40900e1b.png)

But it should not be mandatory to define a cadence target, there are also a lot of workouts without it. It would only be nice to have this option

**Comment by dvmarinoff** [comment]
Got the first demo version of cadence targets running on a preview domain here: [flux-devel.vercel.app](https://flux-devel.vercel.app)

For now it collects cadence targets set with the attributes: Cadence, CadenceResting, OnCadence, OffCadence, meaning
target ranges like 90 to 100 are not yet supported.

I still have no idea how to approach the UI for this feature, and it's super basic for now. Just a small value will show up at the bottom of the cadence field.

**Comment by dvmarinoff** [comment]
Some enhancements from today:

The cadence field when there is no current cadence target:
![Screenshot 2021-12-14 at 18 04 29](https://user-images.githubusercontent.com/7523832/146039831-cc53b991-7861-4353-b24d-85acb8435387.png)

when a cadence target is set:
![Screenshot 2021-12-14 at 18 05 22](https://user-images.githubusercontent.com/7523832/146039846-0caacca8-5597-4379-84b3-80f7f2771dd2.png)

**Comment by TClin76** [comment]
Just come up with an idea.
In addition to Cadence Target, it is also good to have the Power Target shown in %FTP format. 
Since most of structure workout are defined as %FTP for each interval.
Example:
![image](https://user-images.githubusercontent.com/96434118/147913499-3c2f61ca-afd3-4924-8b3b-81558520256a.png)

**Reply by dvmarinoff** [reply]
I do want to add some kind of display for the percentage of FTP and also for the w/kg, but I don't want to make the main grid of data tiles more crowded than it is now. Your proposition brings some balance to the tiles when a cadence target is also shown, but I am afraid in most cases it won't be there unlike the power target which is always present.

In general I plan to develop the instantaneous power graph and put some widgets in the upper part of it that will show FTP percentage, w/kg, pedaling smoothness and power left/right balance. But right now I don't have the time to design those properly.

**Reply by dvmarinoff** [reply]
What I can do in the short run, is to make the target power value switch format on tap. When the User taps the value it will cycle between absolute value (200W) and percentage (80%). And maybe the same for Power but between absolute value and w/kg. Would you like this as temporary solution?

**Reply by TClin76** [reply]
As you mentioned in other threads, speed & distance are meaningless in ERG mode.
In my opinion, absolute power value is so important that it has to be displayed all the time.
So I would recommend a toggle between "**Speed** & **Distance**" and "current **%FTP** & **w/kg**".
For target power, it is a good idea to toggle between absolute value and percentage.

**Reply by dvmarinoff** [reply]
What about having the Target Power field switch on tap between showing just absolute target and absolute target + percentage, as in your suggestion? But I'll have to lose the parens, because of limited of space.

**Reply by TClin76** [reply]
After several runs on Flux these days, I think your earlier solution is better than my suggestion.
Absolute value and percentage toggle while User taps the field, but not showing them in the same time.
Thank u

**Reply by TClin76** [reply]
> What about having the Target Power field switch on tap between showing just absolute target and absolute target + percentage, as in your suggestion? But I'll have to lose the parens, because of limited of space.

Hi @dvmarinoff , is it possible to add this feature? thanks

**Reply by dvmarinoff** [reply]
Sorry for the delay, I completely forgot about this feature. I added some changes today and it is available on the developer version.

<img width="336" alt="Screenshot 2022-03-24 at 12 43 44" src="https://user-images.githubusercontent.com/7523832/159899511-b667cf61-d152-479f-a548-0bc3dc1ef873.png">

**Reply by TClin76** [reply]
Cool. Thank you so much.
